projects/com_protocol/src/main.c

```diff
@@ -0,0 +1,115 @@
+#include "can.h"
+#include "can_ack.h"
+#include "can_msg.h"
+#include "event_queue.h"
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+
+#define CAN_DEVICE_ID 5
+#define CAN_ID_A 0xA
+#define CAN_ID_B 0xB
+#define CAN_PERIOD_A_MS 1000
+#define CAN_PERIOD_B_MS 2000
+
+typedef enum {
+  TEST_CAN_EVENT_RX = 0,
+  TEST_CAN_EVENT_TX,
+  TEST_CAN_EVENT_FAULT,
+} TestCanEvent;
+
+static CanStorage s_can_storage = { 0 };
+
+CanMessage messageA = {
+  .source_id = CAN_DEVICE_ID,
+  .msg_id = CAN_ID_A,
+  .data = 4015,  // random uint16_t becomes uint64_t (not sure how to prevent)
+  .type = CAN_MSG_TYPE_DATA,
+  .dlc = 2,  // 2 bytes in a uint16_t?
+};
+
+CanMessage messageB = {
+  .source_id = CAN_DEVICE_ID,
+  .msg_id = CAN_ID_B,
+  .data = 7777,  // random uint16_t becomes uint64_t (not sure how to prevent)
+  .type = CAN_MSG_TYPE_DATA,
+  .dlc = 2,  // 2 bytes in a uint16_t?
+};
+
+static StatusCode prv_rx_callbackA(const CanMessage *msg, void *context, CanAckStatus *ack_reply) {
+  LOG_DEBUG("Received message with id: 0x%x!\n", msg->msg_id);
+  printf("Data:\n\t %lu", msg->data);
+  // this loop gives me numbers I do not understand
+  // for (uint8_t i = 0; i < msg->dlc; i++) {
+  //   uint8_t byte = 0;
+  //   byte = msg->data >> (i * 8);
+  //   printf("0x%x ", byte);
+  // }

```

Comment:
Depending on the endianness of your system this loop looks like it would successfully read the bytes from the message. You might have more success with `data_u16`.